iree/compiler/Dialect/HAL/Target/MetalSPIRV/BUILD

```diff
@@ -0,0 +1,49 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@rules_foreign_cc//tools/build_defs:cmake.bzl", "cmake_external")
+
+package(
+    default_visibility = ["//visibility:public"],
+    features = ["layering_check"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+cc_library(
+  name = "MetalSPIRV",
+  hdrs = ["MetalSPIRVTarget.h"],
+  srcs = ["MetalSPIRVTarget.cpp"],
+  deps = [
+    ":SPIRVToMSL",
+    "//iree/compiler/Dialect/HAL/Target/SPIRVCommon",
+    "//iree/schemas:metal_executable_def_cc_fbs",
+    "@com_github_google_flatbuffers//:flatbuffers",
+  ]
+)
+
+cc_library(
+  name = "SPIRVToMSL",
+  hdrs = ["SPIRVToMSL.h"],
+  srcs = ["SPIRVToMSL.cpp"],
+  deps = [
+    ":spirv_cross",
+    "@llvm-project//llvm:Support",
+  ],
+)
+
+cmake_external(
+   name = "spirv_cross",
+   lib_source = "@spirv_cross//:all",
+   binaries = ["spirv-cross"],
+)
```

Comment:
We'll probably want to add an overlay `BUILD` file at [build_tools/third_party/](https://github.com/google/iree/tree/main/build_tools/third_party) with a new entry in [WORKSPACE](https://github.com/google/iree/blob/main/WORKSPACE) instead of using `cmake_external`. A `BUILD` file for spirv_cross would be pretty simple, and we have one that can be open sourced easily. `cmake_external` is messier than native support in Bazel for a few reasons.

Reply:
OK, I'll do it.